### PR TITLE
ENCD-5072 Library construction platform display

### PR DESCRIPTION
**Pull request by forresttanaka**

### CODE REVIEW

File: src/encoded/static/components/experiment.js [hunk 1]
```diff
@@ -302,6 +302,46 @@ LibrarySubmitterComments.propTypes = {
...
+        return platforms;
+    }, []);
+
+    if (constructionPlatforms.length > 0) {
+        return (
+            <div data-test="constructionplatform">
+                <dt>Library construction platform</dt>
+                <dd>
+                    {constructionPlatforms.map((platform, i) =>
+                        <React.Fragment key={platform['@id']}>
```
Comment (zoldello) [inline src/encoded/static/components/experiment.js]: Most likely I am wrong; but that has never deterred me from asking... My understanding of React fragments is that it basically tell react now to complain that there no container-tag. That is why you can use <></>. If that is true, why is there a key in the fragment?

File: src/encoded/static/components/experiment.js [hunk 2]
```diff
@@ -302,6 +302,46 @@ LibrarySubmitterComments.propTypes = {
...
+                            {i > 0 ? <span>, </span> : null}
```
Comment (zoldello) [inline src/encoded/static/components/experiment.js]: This reminds me of life under Ecmascript 5. Map creates an array. Would using .join() on the resultant array be sufficient in creating the comma separated list and clearer?

File: src/encoded/static/components/experiment.js [hunk 3]
```diff
@@ -302,6 +302,46 @@ LibrarySubmitterComments.propTypes = {
...
+ * Display, as one element of a "key-value" definition list, all the `construction_platform`
+ * properties in a dataset's replicates' library.
+ */
+const DatasetConstructionPlatform = ({ context }) => {
+    // Collect the library construction platforms from all the dataset replicates.
+    const constructionPlatforms = context.replicates.reduce((platforms, replicate) => {
+        // We easily have duplicates, so only add the replicate library platform if we haven't
+        // already seen it.
+        if ((replicate.library && replicate.library.construction_platform) &&
+            (platforms.findIndex(platform => platform['@id'] === replicate.library.construction_platform['@id']) === -1)) {
```
Comment (zoldello) [inline src/encoded/static/components/experiment.js]: Is it a good idea to put the "platforms.findIndex(platform => platform['@id'] === replicate.library.construction_platform['@id'])" onto its own line. It is the last parenthesis can easily be missed and it easy to think you are doing something like x===y===-1.

File: src/encoded/static/components/experiment.js (hunk 1 above)
Comment (forresttanaka) [reply src/encoded/static/components/experiment.js]: You need the key here also, or else you get (and I just tried it removing that key to be sure):

image

File: src/encoded/static/components/experiment.js (hunk 2 above)
Comment (forresttanaka) [reply src/encoded/static/components/experiment.js]: You cannot join JSX elements. You can only join primitive elements. Other people have other ways to do this, also not involving join, and all of them very difficult to read IMO.

**Pull request merged**